### PR TITLE
[New Rule] Potential Container Escape via Modified notify_on_release File

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/3231
## Summary

This rule detects modification of the cgroup notify_on_release file from inside a container. When the notify_on_release flag is enabled (1) in a cgroup, then whenever the last task in the cgroup exits or attaches to another cgroup, the command specified in the release_agent file is run and invoked from the host. A privileged container with SYS_ADMIN capabilities, enables a threat actor to mount a cgroup directory and modify the notify_on_release flag in order to take advantage of this feature, which could be used for further privilege escalation and container escapes to the host machine.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "OS: Linux",
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where event.module == "cloud_defend" and event.action=="open" and 
+event.type== "change" and file.name: "*notify_on_release"
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]: I'd suggest the same for this one, to use the `file.path: "/sys/fs/cgroup/*notify_on_release` if possible.

Comment (DefSecSentinel) [reply rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]: I agree here better if possible to state the full path to make the query more efficient.

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml (hunk 1 above)
Comment (imays11) [reply rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]: @Aegrah @DefSecSentinel for this attack I was able to create new user and cgroup namespaces in a /tmp directory. 
https://sysdig.com/blog/detecting-mitigating-cve-2022-0492-sysdig/ 
Would you still suggest using file.path with a wildcard to account for unconventional locations? 
image

Comment (imays11) [reply rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]: I'm going to remove the leading wildcard as I did with the release_agent rule, thank you!

File: rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+[metadata]
+creation_date = "2023/10/26"
+integration = ["cloud_defend"]
+maturity = "production"
+min_stack_comments = "New Integration: Cloud Defend"
+min_stack_version = "8.8.0"
+updated_date = "2023/10/26"
```
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:7]:
```suggestion
updated_date = "2023/12/18"
```

File: rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml (hunk 1 above)
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]:
```suggestion
event.type== "change" and file.name: "notify_on_release"
```

File: rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    "OS: Linux",
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where event.module == "cloud_defend" and event.action=="open" and 
+event.type== "change" and file.name: "notify_on_release"
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_potential_container_escape_via_modified_notify_on_release_file.toml:44]:
```suggestion
file where event.module == "cloud_defend" and event.action == "open" and 
event.type == "change" and file.name : "notify_on_release"
```

**Review by Aegrah — APPROVED** [review]

**Pull request merged**